tests/end2end/features/conftest.py

```diff
@@ -200,6 +199,16 @@ def open_path(quteproc, server, path):
     """
     path = path.replace('(port)', str(server.port))
 
+    if os.name == 'nt':
+        # Windows paths need a conversion to a valid URI first. Otherwise we get a
+        # mixture of slashes and backslashes in `path`, resulting in an invalid URL.
+        # `path` already contains `file://` scheme.
+        testdata_path = testutils.abs_datapath().as_uri().replace('file://', '')
```

Comment:
Okay, I think I see what is happening now. `abs_datapath()` returns a proper Windows path but we are going to be inserting it into a file:// URI later. So we've got to convert it into using URI path separators to match.

I still feel a bit weird about manipulating the testdata value in `open_path()`. Like it seems like a generic substitution, what's special about this place that it needs extra work?

Can we just unilaterally URI-isize the path in `abs_datapath()`? Grepping the codebase it seems like it is used to launch userscripts from hints too and I don't think that would work with file:// URIs, so no.

Other options for improvements I can see:

a) only URI-isize it if `path` actually starts with `file://`. The comment states that's the case but it seems like that's an assumption, a true assumption currently. And I think that might make it more clear that we don't have to make this conversion platform dependant?

b) do all of these substitutions in a dedicated method where making it generic and assumption free is a bit more natural. On the other hand, most of these substitutions are only used in one test file so is it worth it?

Reply:
Thanks for the feedback.

Implementing a combination of a) and b) doesn't look like too much effort and it definitely would make things cleaner.

Reply:
@toofar, I gave it a shot in 041de14f9. Would you mind having another look?

Let me know if my explanations are lacking in any area, since I still struggle with these.

Reply:
@toofar, I gave it a shot in 041de14f9. Would you mind having another look?

Let me know if my explanations are lacking in any area, since I still struggle with these.